google-cloud-spanner/src/main/java/com/google/cloud/spanner/AbstractReadContext.java

```diff
@@ -380,6 +383,7 @@ void initTransaction() {
   Span span;
   private final int defaultPrefetchChunks;
   private final QueryOptions defaultQueryOptions;
+  protected boolean routeToLeader = false;
```

Comment:
We actually allow subclasses to change the value depending on the txn type. i.e. If the subclass represents a RW txn context, then overwrite the value with true, otherwise keep it false.

Reply:
In that case I would say that it would be better to define a protected method `protected boolean isRouteToLeader()` that returns `false` in this base class, and override the method in the specific class(es) that should return `true`. That represents the actual situation better than a variable, as it forces the value to be fixed to either true or false depending on the class.

Reply:
Yes that is a good option. Done.